examples/reading_timestamp_test.go

```diff
@@ -0,0 +1,247 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This is a part of the TileDB quickstart tutorial:
+ *   https://docs.tiledb.io/en/latest/quickstart.html
+ *
+ * When run, this program will create a 2D dense array, write data and metadata
+ * to it multiple times keeping track of write timestamps, then read at specific
+ * timestamp using array.OpenAt and prove correlation of timestamps to fragments
+ * and metadata created /updated
+ *
+ */
+
+package examples
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	tiledb "github.com/TileDB-Inc/TileDB-Go"
+)
+
+// Name of array.
+var timestampArrayName = "timestamp_metadata"
+
+func createTimestampArray() {
+	// Create a TileDB context.
+	ctx, err := tiledb.NewContext(nil)
+	checkError(err)
+
+	// The array will be 4x4 with dimensions "rows" and "cols", with domain [1,4].
+	domain, err := tiledb.NewDomain(ctx)
+	checkError(err)
+	rowDim, err := tiledb.NewDimension(ctx, "rows", []int32{1, 4}, int32(4))
+	checkError(err)
+	colDim, err := tiledb.NewDimension(ctx, "cols", []int32{1, 4}, int32(4))
+	checkError(err)
+	err = domain.AddDimensions(rowDim, colDim)
+	checkError(err)
+
+	// The array will be dense.
+	schema, err := tiledb.NewArraySchema(ctx, tiledb.TILEDB_DENSE)
+	err = schema.SetDomain(domain)
+	checkError(err)
+	err = schema.SetCellOrder(tiledb.TILEDB_ROW_MAJOR)
+	checkError(err)
+	err = schema.SetTileOrder(tiledb.TILEDB_ROW_MAJOR)
+	checkError(err)
+
+	// Add a single attribute "a" so each (i,j) cell can store an integer.
+	a, err := tiledb.NewAttribute(ctx, "a", tiledb.TILEDB_INT32)
+	checkError(err)
+	err = schema.AddAttributes(a)
+	checkError(err)
+
+	// Create the (empty) array on disk.
+	array, err := tiledb.NewArray(ctx, timestampArrayName)
+	checkError(err)
+	err = array.Create(schema)
+	checkError(err)
+}
+
+func writeTimestampArray(key string, value string, bias int32) int64 {
+	ctx, err := tiledb.NewContext(nil)
+	checkError(err)
+
+	// Prepare some data for the array
+	data := []int32{
+		1, 2 + bias, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16}
+
+	// Open the array for writing and create the query.
+	array, err := tiledb.NewArray(ctx, timestampArrayName)
+	checkError(err)
+	err = array.Open(tiledb.TILEDB_WRITE)
```

Comment:
You should use OpenAt for the writes too if you want to force a given timestamp and make the test for deterministic.

Reply:
I guess the value passed to timestamp would be `time.Now`. Isn't it the same as just Open as currently written?

Also based on test output behavior seems deterministic even though in the case we 

1. write data + metadata
2. write metadata
3. write metadata

only one fragment is created

I think i am missing something here

Reply:
@Shelnutt2 updated